File: Problems/_beginner_submit_tmp/P7013.cpp
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
public:
    int countNeedUpgrade(vector<int>& versions, int baseline) {
        // 排序后用 lower_bound 找第一个 >= baseline
        vector<int> a = versions;
        sort(a.begin(), a.end());
        auto it = lower_bound(a.begin(), a.end(), baseline);
        // 距离末尾即为答案个数
        return (int)(a.end() - it);
    }
};

File: Problems/_beginner_submit_tmp/P7013.java
import java.util.*;

public class Solution {
    public int countNeedUpgrade(int[] versions, int baseline) {
        // 复制后排序，便于二分
        int n = versions.length;
        int[] a = Arrays.copyOf(versions, n);
        Arrays.sort(a);
        // lower_bound：第一个 >= baseline 的下标
        int lo = 0, hi = n;
        while (lo < hi) {
            int mid = (lo + hi) >>> 1;
            if (a[mid] < baseline) {
                lo = mid + 1;
            } else {
                hi = mid;
            }
        }
        // 从 lo 到末尾均需巡检
        return n - lo;
    }
}
